### PR TITLE
remove Centos 8 from Helix as it is EOL

### PR DESCRIPTION
**Pull request by wfurt**
It is EOL https://www.centos.org/centos-linux-eol/ as December 31st, 2021 and we no longer produce docker images https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/622

contributes to  #75262

cc: @tmds @omajid @danmoseley

### CODE REVIEW

**Comment by karelz** [comment]
Just curious if we should start testing CentOS Stream instead, or if coverage of Fedora + RHEL is sufficient in the space ... any thoughts @wfurt?

**Comment by wfurt** [comment]
Some background reading: https://thenewstack.io/centos-9-stream-is-now-available-but-should-you-use-it/

Seems pretty close to Fedora (34) ->  may not need separate run. 
```
CentOS Stream is a continuous-delivery distribution serving as the next point-release of RHEL.
```
so this is going be ongoing battle. This may work great for live systems but it seems like this is going to be more tricky with container rebuilds and manual updates in runtime. 

Perhaps  @tmds @omajid and comment on what would give us best coverage while minimizing runs.

**Comment by omajid** [comment]
A couple of thoughts.

- The CentOS 8 leg, in my mind, was kind of a stand-in for RHEL 8. We have RHEL 7 in the matrix (`RedHat.7.Amd64.Open`) but not 8. If we can add RHEL 8 directly (or perhaps another RHEL rebuild like CentoS 8 was?), we can probably do without an explicit CentOS 8 leg.
- Fedora 34 is itself EOL: https://docs.fedoraproject.org/en-US/releases/eol/. We will remove that at some point soon too, right? Besides, RHEL 8 was, roughly, a fork from Fedora 28. There's enough changes between Fedora 28 and 34 that I wouldn't trust test result from 34 to tell me what the compatibility situation with 28 is.
- CentOS Stream 8 tracks the upcoming version of RHEL 8. It's probably easier to think of it as"this will be in the next update of RHEL 8" as in "when users run `dnf update` in about 6 months, this is what they will get". That wouldn't be a major version upgrade. There won't be any API or ABI change in major components or core libraries. Generally speaking, breaking changes would not be allowed. This isn't an update to RHEL 9. So I wouldn't expect any breakage here and the maintenance work should be little-to-none. If we see any issues, we have time to fix it as a bug in RHEL and/or update .NET to handle that before it starts affecting folks using RHEL (or a rebuild) in production.

In other words, we should add RHEL 8, or a RHEL-8-rebuild in our CI matrix. If not, CentOS Stream 8 shouldn't be too bad.

> This may work great for live systems but it seems like this is going to be more tricky with container rebuilds and manual updates in runtime.

I am not sure I follow this. Could you elaborate?

**Review by rzikm — APPROVED** [review]
